m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -0,0 +1,745 @@
+                gaugeValue.setDValue(Double.MAX_VALUE);
+                metricValue.setGauge(gaugeValue);
+                break;
+            case TIMER:
+                TimerValue timerValue = resourcePool.getTimerValue();
+                timerValue.setI64Value(Long.MAX_VALUE);
+                metricValue.setTimer(timerValue);
+                break;
+        }
+
+        metric.setMetricValue(metricValue);
+
+        return metric;
+    }
+
+    private Set<MetricTag> addMetricTagsToSet(Map<String, String> tags) {
+        Set<MetricTag> metricTagSet = resourcePool.getTagList();
+
+        if (tags == null) {
+            return metricTagSet;
+        }
+
+        for (Map.Entry<String, String> tag : tags.entrySet()) {
+            metricTagSet.add(createMetricTag(tag.getKey(), tag.getValue()));
+        }
+
+        return metricTagSet;
+    }
+
+    private MetricTag createMetricTag(String tagName, String tagValue) {
+        MetricTag metricTag = resourcePool.getMetricTag();
+        metricTag.setTagName(tagName);
+
+        if (tagValue != null && !tagValue.isEmpty()) {
+            metricTag.setTagValue(tagValue);
+        }
+
+        return metricTag;
+    }
+
+    private int calculateSize(Metric metric) {
+        try {
+            synchronized (calcLock) {
+                metric.write(calcProtocol);
+
+                return calc.getSizeAndReset();
+            }
+        } catch (TException e) {
+            return DEFAULT_METRIC_SIZE;
+        }
+    }
+
+    private void reportCopyMetric(
+        Metric metric,
+        int size,
+        MetricType metricType,
+        long iValue,
+        double dValue
+    ) {
+        Metric metricCopy = resourcePool.getMetric();
+        metricCopy.setName(metric.getName());
+        metricCopy.setTags(metric.getTags());
+        // Unfortunately, there is no finer time resolution for Java 7
+        metricCopy.setTimestamp(System.currentTimeMillis() * Duration.NANOS_PER_MILLI);
+        metricCopy.setMetricValue(resourcePool.getMetricValue());
+
+        switch (metricType) {
+            case COUNTER:
+                CountValue countValue = resourcePool.getCountValue();
+                countValue.setI64Value(iValue);
+                metricCopy.getMetricValue().setCount(countValue);
+                break;
+            case GAUGE:
+                GaugeValue gaugeValue = resourcePool.getGaugeValue();
+                gaugeValue.setDValue(dValue);
+                metricCopy.getMetricValue().setGauge(gaugeValue);
+                break;
+            case TIMER:
+                TimerValue timerValue = resourcePool.getTimerValue();
+                timerValue.setI64Value(iValue);
+                metricCopy.getMetricValue().setTimer(timerValue);
+                break;
+        }
+
+        try {
+            metricQueue.put(new SizedMetric(metricCopy, size));
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't remember the semantics of the current java client when this happens, does it throw an exception too? 

Might it be better to log instead of throwing here?

Reply:
Looks like we just drop it on the floor ([here](https://github.com/uber-go/tally/blob/master/m3/reporter.go#L438-L441)).

I'll log it.